test/specs/lib/AutoControlledComponent-test.js

```diff
@@ -32,6 +32,11 @@ describe('extending AutoControlledComponent', () => {
     TestClass = createTestClass({ autoControlledProps: [], state: {} })
   })
 
+  it('does not throw with a `null` state', () => {
```

Comment:
Extra test to catch the edge case I ran into.  If you have `state = undefined` React will initialize state as `null`.  This then throws as we look for keys on `null`.  I added a comment about this and a check for it in `getAutoControlledStateValue`.